src/api/goods.ts

```diff
@@ -5,15 +5,26 @@ const API_URL = `https://mate-academy.github.io/react_dynamic-list-of-goods/good
 
 export function getAll(): Promise<Good[]> {
   return fetch(API_URL)
-    .then(response => response.json());
+    .then(response => {
+      if (!response.ok) {
+        return Promise.reject(
+          new Error(response.statusText),
+        );
+      }
+
+      return response.json();
+    })
+    .catch(error => new Error(error));
```

Comment:
Is there any sense to throw an error if error is already thrown (we catching it here)
Instead of this - create state for error and show error message to users
(unfortunately users don't open console to see if there any error while loading data, they like to see information about it on UI)